examples/image-search/search.cpp

```diff
@@ -27,59 +30,89 @@ void my_print_help(int argc, char **argv, my_app_params &params) {
 }
 
 // returns success
-bool my_app_params_parse(int argc, char **argv, my_app_params &params) {
+bool my_app_params_parse(int argc, char **argv, my_app_params &params)
+{
     bool invalid_param = false;
-    for (int i = 1; i < argc; i++) {
+    for (int i = 1; i < argc; i++)
+    {
         std::string arg = argv[i];
 
-        if (arg == "-m" || arg == "--model") {
-            if (++i >= argc) {
+        if (arg == "-m" || arg == "--model")
+        {
+            if (++i >= argc)
+            {
                 invalid_param = true;
                 break;
             }
             params.model = argv[i];
-        } else if (arg == "-t" || arg == "--threads") {
-            if (++i >= argc) {
+        }
+        else if (arg == "-t" || arg == "--threads")
+        {
+            if (++i >= argc)
+            {
                 invalid_param = true;
                 break;
             }
             params.n_threads = std::stoi(argv[i]);
-        } else if (arg == "-n" || arg == "--results") {
-            if (++i >= argc) {
+        }
+        else if (arg == "-n" || arg == "--results")
+        {
+            if (++i >= argc)
+            {
                 invalid_param = true;
                 break;
             }
             params.n_results = std::stoi(argv[i]);
-        } else if (arg == "-v" || arg == "--verbose") {
-            if (++i >= argc) {
+        }
+        else if (arg == "-v" || arg == "--verbose")
+        {
+            if (++i >= argc)
+            {
                 invalid_param = true;
                 break;
             }
             params.verbose = std::stoi(argv[i]);
-        } else if (arg == "-h" || arg == "--help") {
+        }
+        else if (arg == "-h" || arg == "--help")
+        {
             my_print_help(argc, argv, params);
             exit(0);
-        } else if (arg.starts_with('-')) {
-            if (i != 0) {
+        }
+        else if (arg.find('-') == 0)
+        {
+            if (i != 0)
+            {
                 printf("%s: unrecognized argument: %s\n", __func__, arg.c_str());
                 return false;
             }
-        } else {
+        }
+        else
+        {
             // assume search string from here on out
-            params.search_text = arg;
-            for (++i; i < argc; i++) {
-                params.search_text += " ";
-                params.search_text += argv[i];
+            if (i == argc - 1 && is_image_file_extension(arg))
+            {
+                params.img_path = arg;
+            }
+            else
+            {
+                params.search_text = arg;
+                for (++i; i < argc; i++)
```

Comment:
you kept the ++i i++ :rofl:

Reply:
:D even the compiler should be surprised by it